### PR TITLE
fix: 8878 Data not parsed as float in google sheets

### LINKED ISSUE #8878: [Bug]: Data not being parsed into Float
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When the user navigates into Google API and sets the Query row into 1-100 which has the combination of both empty decimal and integers. The API displays an error
Screenshot 2021-10-28 at 12 22 06 PM

### Steps To Reproduce

1) Navigate to Google sheet
2) Authorise the sheets 
3) Click on the new API 
4) Select fetch as the method
5) Add the where clause 
6) Click on Run 
and observe

Ensure the user data has a combination of integer and decimal in the google sheet 

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by leotom2000**
Data not parsed as float when fetching in Google sheet

## Description

When fetching data from Google Sheets, specific column had both number and decimals.
when a decimal key is used to fetch, the validation happens correct by identifying the data type
But the identified data type is not assigned for further processing
The fix involves assigning the correct identified datatype for further processing

Fixes #8878
 

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Not a new feature
- Not a Breaking change
- No documentation update is required 

## How Has This Been Tested?

* Unit Test is not added
* Verified with google sheet data
 
## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/8878-GS_Data_not_parsed_float_issue 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.05 **(0.01)** | 36.57 **(0.01)** | 34.49 **(0)** | 55.56 **(0)**
 :green_circle: | app/client/src/utils/WorkerUtil.ts | 89.76 **(0)** | 72.55 **(1.96)** | 100 **(0)** | 93.33 **(0)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**</details>

### CODE REVIEW

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/ce/FilterDataServiceCE.java [hunk 1]
```diff
@@ -605,6 +605,8 @@ private PreparedStatement setValueInStatement(PreparedStatement preparedStatemen
             DataType inputDataType = stringToKnownDataTypeConverter(value);
             if (DataType.NULL.equals(inputDataType)) {
                 dataType = DataType.NULL;
+            }else{
```
Comment (mohanarpit) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/ce/FilterDataServiceCE.java:608]: Nit: Please format the code before committing & pushing. We aim to have consistent formatting of the entire codebase.

**Comment by nidhi-nair** [comment]
@leotom2000 Is it possible to add a test case to this flow so we're sure any future refactor takes this use case into account?

**Comment by leotom2000** [comment]
PR status is in progress, one environment issue is preventing to test and close this issue

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/ce/FilterDataServiceCE.java (hunk 1 above)
Comment (leotom2000) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/ce/FilterDataServiceCE.java]: Code format updated

**Comment by leotom2000** [comment]
> @leotom2000 Is it possible to add a test case to this flow so we're sure any future refactor takes this use case into account?

Added Unit test case

File: app/server/appsmith-interfaces/src/test/java/com/appsmith/external/services/FilterDataServiceTest.java [hunk 2]
```diff
@@ -767,6 +767,64 @@ public void testFilterInConditionForStringsNew() {
         }
     }
 
+    @Test
+    public void testFilterInConditionForNumbersAndDecimals() {
```
Comment (nidhi-nair) [inline app/server/appsmith-interfaces/src/test/java/com/appsmith/external/services/FilterDataServiceTest.java:771]: How is this test different from `testFilterInConditionForNumbersNew`

Comment (leotom2000) [reply app/server/appsmith-interfaces/src/test/java/com/appsmith/external/services/FilterDataServiceTest.java:771]: Here it is a mix of Numbers and Decimals, the issue come when the datatype to be filtered is different.
Thant is handled and tested in unit test

Comment (nidhi-nair) [reply app/server/appsmith-interfaces/src/test/java/com/appsmith/external/services/FilterDataServiceTest.java:771]: Understood, could you please add this  as a comment to the test, it'll make it easier for future devs to understand the intent.

Comment (leotom2000) [reply app/server/appsmith-interfaces/src/test/java/com/appsmith/external/services/FilterDataServiceTest.java:776]: Added code comments and few more data for testing for unit testing

**Review by nidhi-nair — APPROVED** [review]

**Comment by leotom2000** [comment]
Git Pull and Git Push, have created a lot of unrelated changes.
So not planning to merge this pull request,
will discuss and create another pull request with the known changes

**Comment by leotom2000** [comment]
This PR can be closed without merging, since the PR #9712 for issues Fixes #9711, Fixes #9399,Fixes #9397 has resolved and merged the changes which was handled by this PR. Tested the prod environment and found the issue handled by the code changes for this PR is already resolved.

**Comment by leotom2000** [comment]
Closing this PR since the issue was indirectly resolved by PR #9712

**Pull request closed without merging**